src/frontend/src/pages/SettingsPage/UserSecureSettings/UserSecureSettingsEdit.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * This file is part of NER's FinishLine and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useForm } from 'react-hook-form';
+import { Grid, FormControl, FormLabel } from '@mui/material';
+import ReactHookTextField from '../../../components/ReactHookTextField';
+import { SecureSettingsFormInput } from './UserSecureSettings';
+
+interface UserSecureSettingsEditProps {
+  currentSettings: SecureSettingsFormInput;
+  onSubmit: (data: SecureSettingsFormInput) => Promise<void>;
+}
+
+const schema = yup.object().shape({
+  street: yup.string().required('Street is required'),
+  city: yup.string().required('City is required'),
+  state: yup.string().required('State is required'),
+  zipcode: yup.string().required('Zipcode is required'),
+  phoneNumber: yup.string().required('Phone number is required'),
```

Comment:
self-reminder: make a ticket to make this a better input/ validation than just a textbox